File: Baekjoon/S5/C#/S5_다리 놓기.cs
https://www.acmicpc.net/problem/1010

using System;

namespace Baekjoon
{
    class Program
    {
        static int[,] dp = new int[31, 31];

        static void Main(string[] args)
        {
            int t = int.Parse(Console.ReadLine());
            for (int i = 1; i <= 30; i++)
                dp[1, i] = i;

            for (int i = 2; i <= 30; i++)
            {
                for (int j = i; j <= 30; j++)
                {
                    for (int k = j - 1; k >= 1; k--)
                    {
                        dp[i, j] += dp[i - 1, k];
                    }
                }
            }

            while (t-- > 0)
            {
                string[] str = Console.ReadLine().Split();
                Console.WriteLine(dp[int.Parse(str[0]), int.Parse(str[1])]);
            }
        }
    }
}

File: Baekjoon/S5/JAVA/S5_다리 놓기.java
//https://www.acmicpc.net/problem/1010

import java.io.*;
import java.util.StringTokenizer;

public class Main{
    public static void main(String[] args) throws IOException{
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	    StringBuilder sb = new StringBuilder();
	    
	    int t = Integer.parseInt(br.readLine());
	    while(t-- > 0){
	        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
			int n = Integer.parseInt(st.nextToken());
			int m = Integer.parseInt(st.nextToken());

            int[][] dp = new int[n+1][m+1];
				
            for(int i = 1; i <= m; i++)
                dp[1][i] = i;
            
            for(int i = 2; i <= n; i++){
                for(int j = 2; j <= m; j++){
                    dp[i][j] = dp[i][j-1] + dp[i-1][j-1];
                }
            }
            
			sb.append(dp[n][m]+"\n");
	    }
	    
	    System.out.print(sb);
	    br.close();
	}
}
